File: backend/app/schemas/history.py
from datetime import datetime

from app.schemas.common import ApiResponse, PageData, SchemaBase


class StoreSnapshot(SchemaBase):
    id: str
    store_code: str
    name: str
    category: str
    address: str
    area: str = ""
    image_url: str | None = None
    is_available: bool


class HistoryItem(SchemaBase):
    id: str
    action: str
    occurred_at: datetime
    store: StoreSnapshot


HistoryListResponse = ApiResponse[PageData[HistoryItem]]

File: frontend/src/api/history.ts
import { get } from './client'
import type { PageData } from './stores'

export interface HistoryStoreSnapshot {
  id: string
  storeCode: string
  name: string
  category: string
  address: string
  area: string
  imageUrl?: string | null
  isAvailable: boolean
}

export interface HistoryRecord {
  id: string
  action: 'RANDOM_PICK' | 'DETAIL_VIEW' | string
  occurredAt: string
  store: HistoryStoreSnapshot
}

interface HistoryQuery {
  action?: string
  page: number
  page_size: number
}

export function getHistory(page = 1, pageSize = 100): Promise<PageData<HistoryRecord>> {
  return get<PageData<HistoryRecord>, HistoryQuery>('/me/history', { page, page_size: pageSize })
}
